controller/setting_controller.go

```diff
@@ -1387,7 +1387,12 @@ func (info *ClusterInfo) collectSettings() error {
 
 		// Setting that should be collected as boolean (true if configured, false if not)
 		case includeAsBoolean[settingName]:
-			settingMap[setting.Name] = setting.Value != ""
+			definition, ok := types.GetSettingDefinition(types.SettingName(setting.Name))
+			if !ok {
+				logrus.WithError(err).Warnf("Failed to get Setting %v definition", setting.Name)
+				continue
```

Comment:
A quick question. Can this case ever happen? I think the setting is already in `includeAsBoolean` which means it should already be in `GetSettingDefinition`, shouldn't it?

Reply:
I don't think having the setting in the includeAsBoolean guaranteed its presence in the GetSettingDefinition.  Although this case is unlikely to occur, we should still do a null check to avoid any potential NPE.

Reply:
I think at least it can avoid the NPE issue when we somehow typo or miss the entries of `settingDefinitions`.

Reply:
Sounds good. Thanks all